### PR TITLE
Push to test Pypi

### PR DESCRIPTION
**Pull request by giorgiosironi**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,4 +1,25 @@
...
+    return "[${sectionName}]\nusername=${credentials.username}\npassword=${credentials.password}"
+}
+
+def withPypiCredentials(String env, String sectionName, doSomething) {
+    try {
+        writeFile(file: '.pypirc', text: jsonToPypirc(sh(
+            script: "vault.sh kv get -format=json secret/containers/pypi/${env} | jq .data.data",
+            returnStdout: true
+        ).trim(), sectionName))
+        doSomething()
```
Comment (giorgiosironi) [inline Jenkinsfile]: the convention in Groovy for these closures is usually `it()`

Comment (de-code) [reply Jenkinsfile]: What does `it` stand for? Would suggest to me something like iterator or item (when iterating). Are you aware of a documentation of that convention?

**Comment by de-code** [comment]
This seems to work now.

Creating the version is a bit hacky. pypi, including test pypi doesn't allow replacing a version. Therefore we need to create a unique id. Unfortunately the version has to follow certain criteria which prevents us from just using the commit in full or in part. It is now converting the last part of the commit to an integer and adds it to a date based version. e.g. `2019.5.9.dev74639288`. Maybe a random number would do as well. That only applies for the test release.

For the actual release it's just using the version from the tag. No hacking required there. (I haven't tested that of course, but hopefully it's working, given it's passing through some of the same code / script branches)

**Comment by de-code** [comment]
The version identification formation is specified by [PEP 440](https://www.python.org/dev/peps/pep-0440/)

**Pull request merged**